From f9a9502e0ae9da2805f64d11e9d3569e95a0d5c9 Mon Sep 17 00:00:00 2001
From: Nitin Garg <nitin.garg@freescale.com>
Date: Wed, 12 Dec 2012 14:40:19 -0600
Subject: [PATCH 014/231] ENGR00236834: RTC fix a crash in mxc_rtc_read_time
 while registering rtc

Set pdata (before calling rtc_device_register) as its required by
rtc_device_register as it checks if alarm is already set,
calling mxc_rtc_read_time.

Signed-off-by: Nitin Garg <nitin.garg@freescale.com>
---
 drivers/rtc/rtc-mxc_v2.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/rtc/rtc-mxc_v2.c b/drivers/rtc/rtc-mxc_v2.c
index 6b02c92..fd2629d 100644
--- a/drivers/rtc/rtc-mxc_v2.c
+++ b/drivers/rtc/rtc-mxc_v2.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2004-2011 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright (C) 2004-2012 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -631,6 +631,10 @@ static int mxc_rtc_probe(struct platform_device *pdev)
 	__raw_writel(0xFFFFFFFF, ioaddr + SRTC_LPSR);
 	udelay(100);
 
+	/* Set pdata as its required by rtc_device_register */
+	/* as it checks if alarm is already set, calling mxc_rtc_read_time */
+	platform_set_drvdata(pdev, pdata);
+
 	rtc = rtc_device_register(pdev->name, &pdev->dev,
 				  &mxc_rtc_ops, THIS_MODULE);
 	if (IS_ERR(rtc)) {
-- 
1.7.10.4

